cpp/include/cudf/aggregation.hpp

```diff
@@ -599,6 +609,255 @@ std::unique_ptr<Base> make_udf_aggregation(udf_type type,
                                            std::string const& user_defined_aggregator,
                                            data_type output_type);
 
+/**
+ * @brief The interface for host-based UDF implementation.
+ *
+ * An implementation of host-based UDF needs to be derived from this base class, defining
+ * its own version of the required functions. In particular:
+ *  - The derived class is required to implement `get_empty_output`, `operator()`, `is_equal`,
+ *    and `clone` functions.
+ *  - If necessary, the derived class can also override `do_hash` to compute hashing for its
+ *    instance, and `get_required_data` to selectively access to the input data as well as
+ *    intermediate data provided by libcudf.
```

Comment:
How well does this format in the output doxygen?
Also, how difficult would it be to include a simple example here of a derived class? It it is too much maybe include a link to one of the tests examples?

Reply:
Format should be fine. I've added a simple example in  the doxygen.